x-pack/legacy/plugins/uptime/server/rest_api/overview_filters/get_overview_filters.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { UMServerLibs } from '../../lib/lib';
+import { UMRestApiRouteFactory } from '../types';
+import { objectValuesToArrays } from '../../lib/helper';
+
+const arrayOrStringType = schema.maybe(
+  schema.oneOf([schema.string(), schema.arrayOf(schema.string())])
+);
+
+export const createGetOverviewFilters: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
+  method: 'GET',
+  path: '/api/uptime/filters',
+  validate: {
+    query: schema.object({
+      dateRangeStart: schema.string(),
+      dateRangeEnd: schema.string(),
+      search: schema.maybe(schema.string()),
+      locations: arrayOrStringType,
+      schemes: arrayOrStringType,
+      ports: arrayOrStringType,
+      tags: arrayOrStringType,
+    }),
+  },
+
+  options: {
+    tags: ['access:uptime'],
+  },
+  handler: async ({ callES }, _context, request, response) => {
+    const { dateRangeStart, dateRangeEnd, locations, schemes, search, ports, tags } = request.query;
+
+    let parsedSearch: Record<string, any> | undefined;
+    if (search) {
+      try {
+        parsedSearch = JSON.parse(search);
+      } catch (e) {
+        return response.badRequest();
```

Comment:
We really need to log or return the error here to help with debugging, even if only for our own dev experience. Swallowing errors should basically never be something we do, and if it is it probably needs a very long comment justifying why. 

Since the error is in the request content, we should return this to the client. I'd extract the error message and set it as the response body for `response.badRequest({body: e.toString()})` or similar.

Reply:
Yeah I 100% agree, this is really lazy code. Edited in https://github.com/elastic/kibana/pull/50837/commits/a1bcd077e25159969b71e1ce8d6bde0c4ffeecda.